sklearn/feature_extraction/tests/test_text.py

```diff
@@ -995,3 +996,29 @@ def test_vectorizer_string_object_as_input():
             ValueError, message, vec.fit, "hello world!")
         assert_raise_message(
             ValueError, message, vec.transform, "hello world!")
+
+
+@mark.parametrize("vec, index", [
+        (HashingVectorizer(ngram_range=(2, 1)), 0),
+        (CountVectorizer(ngram_range=(2, 1)), 1),
+        (TfidfVectorizer(ngram_range=(2, 1)), 2)
+    ])
+def test_vectorizers_invalid_ngram_range(vec, index):
+    # vectorizers could be initialized with invalid ngram range
+    # test for raising error message
+    invalid_range = vec.ngram_range
+    message = ("Invalid value for ngram_range=%s "
+               "lower boundary larger than the upper boundary"
+               % str(invalid_range))
+
+    # HashingVectorizer implements fit, transform
+    # CountVectorizer, TfidfVectorizer implement fit, fit_transform
```

Comment:
That's not true, all have fit, transform and fit_transform.

The issue is that one can't run transform, if the estimator wasn't fitted.

We can just use,
```py
if isinstance(vect, HashingVectorizer):
    assert_..
```
to be more explicit.
And remove the `index` object altogether to not having to think about the index mapping.

We can tests `fit`, `fit_transform` for all and `transform` just for `HashingVectorizer`